collectors/python.d.plugin/edgex/edgex.chart.py

```diff
@@ -0,0 +1,312 @@
+            port = self.edgex_ports['core_data']
+        )
+        # self.url_core_data_events = '{url}:{port}/api/v1/'.format(
+        #     url = self.url,
+        #     port = self.edgex_ports['core_data']
+        # )
+        self.url_core_metadata = '{url}:{port}/api/v1/'.format(
+            url = self.url,
+            port = self.edgex_ports['core_metadata']
+        )
+        self.url_core_command = '{url}:{port}/api/v1/'.format(
+            url = self.url,
+            port = self.edgex_ports['core_command']
+        )
+        # self.url_sys_mgmt = '{url}:{port}/api/v1/'.format(
+        #     url = self.url,
+        #     port = self.edgex_ports['sys_mgmt']
+        # )
+        self.url_support_logging = '{url}:{port}/api/v1/'.format(
+            url = self.url,
+            port = self.edgex_ports['support_logging']
+        )
+
+
+    def check(self):
+        self.methods = [
+            METHODS(
+                get_data=self._get_core_throughput_events,
+                url=self.url_core_data,
+                run=self.configuration.get('events_per_second', True),
+            ),
+            METHODS(
+                get_data=self._get_core_throughput_readings,
+                url=self.url_core_data,
+                run=self.configuration.get('events_per_second', True)
+            ),
+            METHODS(
+                get_data=self._get_device_info,
+                url=self.url_core_metadata,
+                run=self.configuration.get('number_of_devices', True)
+            ),
+            METHODS(
+                get_data=self._get_metrics_data,
+                url=self.url_core_metadata,
+                run=self.configuration.get('metrics', True)
+            ),
+            METHODS(
+                get_data=self._get_metrics_data,
+                url=self.url_core_data,
+                run=self.configuration.get('metrics', True)
+            ),
+            METHODS(
+                get_data=self._get_metrics_data,
+                url=self.url_core_command,
+                run=self.configuration.get('metrics', True)
+            ),
+            METHODS(
+                get_data=self._get_metrics_data,
+                url=self.url_support_logging,
+                run=self.configuration.get('metrics', True)
+            )
+            # METHODS(
+            #     get_data=self._get_metrics_data
+            #     url=self.url_sys_mgmt,
+            #     run=self.configuration.get('metrics', True)
+            # )
+         ]
+        return UrlService.check(self)
+
+    def _get_data(self):
+        threads = list()
+        queue = Queue()
+        result = {}
+        for method in self.methods:
+            if not method.run:
+                continue
+            th = threading.Thread(
+                target=method.get_data,
+                args=(queue, method.url),
+            )
+            th.daemon = True
+            th.start()
+            threads.append(th)
```

Comment:
This can be simplified to:

```python
if method.run:
    th = threading.Thread(
        target=method.get_data,
        args=(queue, method.url),
    )
    th.daemon = True
    th.start()
    threads.append(th)
```

Reply:
This was taken from the elasticsearch example. Should I commit a change to that as well?

Reply:
Possibly. I'd like to see what @ilyam8 says about it if it's not completely new.

Not having the negated condition in the if statement technically makes things a few cycles faster and arguably makes it clearer what's going on (if `method.run` is not None or false, do something to process it's data), but code consistency is important.

Reply:
No

Reply:
And it doesnt look like simplification to me